modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/AdminTooltip.js

```diff
@@ -0,0 +1,219 @@
+		variables: {
+			name: sku,
+		},
+	});
+
+	const cardStyle = {
+		left: showTooltip.details.cx,
+		top: showTooltip.details.cy,
+	};
+
+	return (
+		<ClayCard className="admin-tooltip" style={cardStyle}>
+			<ClayCard.Body className="row">
+				<ClayForm.Group className="col-12 form-group-sm text-left">
+					<label htmlFor={`${namespace}pin-position`}>
+						{Liferay.Language.get('position')}
+					</label>
+					<ClayInput
+						id={`${namespace}pin-position`}
+						onChange={(event) => setPinLabel(event.target.value)}
+						placeholder={Liferay.Language.get(
+							'insert-your-name-here'
+						)}
+						type="text"
+						value={pinLabel}
+					/>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-12 form-group-sm">
+					<ClayRadioGroup
+						className="d-flex justify-content-start mt-4"
+						inline
+						onSelectedValueChange={(val) => setLinkedValue(val)}
+						selectedValue={linkedValue}
+					>
+						<ClayRadio label="Linked to SKU" value="sku" />
+						<ClayRadio label="Linked to Diagram" value="diagram" />
+					</ClayRadioGroup>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-9 form-group-sm text-left">
+					<label htmlFor={`${namespace}pin-sku`}>
+						{Liferay.Language.get('select-sku')}
+					</label>
+					<ClayAutocomplete>
+						<ClayAutocomplete.Input
+							id={`${namespace}pin-sku`}
+							onChange={(event) => setSku(event.target.value)}
+							placeholder={Liferay.Language.get('type-sku-here')}
+							value={sku}
+						/>
+						<ClayAutocomplete.DropDown
+							active={(!!resource && !!sku) || initialLoading}
+						>
+							<ClayDropDown.ItemList>
+								{(error || (resource && resource.error)) && (
+									<ClayDropDown.Item className="disabled">
+										No Results Found
+									</ClayDropDown.Item>
+								)}
+								{!error &&
+									resource &&
+									resource.results &&
+									resource.results.map((item) => (
+										<ClayAutocomplete.Item
+											key={item.id}
+											match={sku}
+											value={item.name}
+										/>
+									))}
+							</ClayDropDown.ItemList>
+						</ClayAutocomplete.DropDown>
+						{loadingAd && <ClayAutocomplete.LoadingIndicator />}
+					</ClayAutocomplete>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-3 form-group-sm">
+					<label htmlFor={`${namespace}pin-quantity`}>
+						{Liferay.Language.get('quantity')}
+					</label>
+					<ClayInput
+						id={`${namespace}pin-quantity`}
+						onChange={(event) =>
+							setQuantity(parseInt(event.target.value, 10))
+						}
+						type="number"
+						value={quantity}
+					/>
+				</ClayForm.Group>
+
+				<ClayForm.Group className="col-6 d-flex form-group-sm justify-content-start mt-4">
+					<ClayButton
+						displayType="link"
+						onClick={() => {
+							setRemovePinHandler({
+								handler: true,
+								pin: showTooltip.details.id,
+							});
+							setShowTooltip({
+								details: {
+									cx: null,
+									cy: null,
```

Comment:
do you need this explicit nulls? is the field optional (and can be marked with ?)